src/Sulu/Bundle/WebsiteBundle/Routing/ContentRouteProvider.php

```diff
@@ -170,6 +178,15 @@ public function getRouteCollectionForRequest(Request $request)
             } elseif (!$this->checkResourceLocator($resourceLocator, $prefix)) {
                 return $collection;
             } else {
+                if ($document instanceof ExtensionBehavior) {
+                    $documentSegmentKey = $document->getExtensionsData()['excerpt']['segment'];
+                    $segment = $this->requestAnalyzer->getSegment();
+
+                    if ($segment && $segment->getKey() !== $documentSegmentKey) {
```

Comment:
either we need to check if the `$documentSegmentKey` is not null, or we need to adjust the `requestAnalyzer->changeSegment()` to accept null as parameter 🙂

Reply:
Is this the error you have talked about today? I can't really reproduce it, how have you managed that?

Reply:
No idea why it wasn't happening before :see_no_evil: Fixed it now in my last commit, the segment does not change if no segment is set on the loaded document.